packages/tooltip/src/use-tooltip.ts

```diff
@@ -97,31 +97,31 @@ export function useTooltip(props: UseTooltipProps = {}) {
   const exitTimeoutRef = React.useRef<NodeJS.Timeout>()
 
   const openWithDelay = () => {
-    enterTimeoutRef.current = setTimeout(onOpen, showDelay)
+    enterTimeoutRef.current = setTimeout(_onOpen, showDelay)
   }
 
   const closeWithDelay = () => {
     if (enterTimeoutRef.current) {
       clearTimeout(enterTimeoutRef.current)
     }
-    exitTimeoutRef.current = setTimeout(onClose, hideDelay)
+    exitTimeoutRef.current = setTimeout(_onClose, hideDelay)
   }
 
   const onClick = () => {
     if (closeOnClick) {
-      onClose()
+      closeWithDelay()
```

Comment:
I ran into a bug caused by delay open and close immediately,
![2020-07-04 at 11 53 16 - Lavender Alpaca](https://user-images.githubusercontent.com/39694575/86506457-1676ba80-bded-11ea-8007-e668849f6d66.gif)

Fixed it by changing the `onClose` to `closeWithDelay` on all handlers, since that clears out the `open timer`.
This is caused because of the even propagation from mousedown to focus on click.